src/platform/site-wide/promoBanners/components/App/index.js

```diff
@@ -1,55 +0,0 @@
-// Node modules.
-import React, { useState } from 'react';
-import PromoBanner from '@department-of-veterans-affairs/component-library/PromoBanner';
-import PropTypes from 'prop-types';
-
-const DISMISSED_PROMO_BANNERS_LOCAL_STORAGE_KEY = 'dismissedPromoBanners';
-
-export const App = ({ href, id, text, type }) => {
-  // Derive dismissed promo banners.
-  const dismissedPromoBannerIDs =
-    JSON.parse(
-      localStorage.getItem(DISMISSED_PROMO_BANNERS_LOCAL_STORAGE_KEY),
-    ) || [];
-
-  // Derive whether the promo banner is dismissed.
-  const [isDismissed, setIsDismissed] = useState(
-    dismissedPromoBannerIDs.includes(id),
-  );
-
-  // On dismiss, add the promo banner to local storage.
-  const onClose = () => {
-    // Remove class that pads the site footer
-    document.body.classList.remove('va-pad-promo-banner');
```

Comment:
We need this body class toggled whenever the promo banner is present. Without it we have a padding issue in the footer with the banner covering content and the only way to view is to dismiss the banner. This would revert [https://github.com/department-of-veterans-affairs/vets-website/pull/20435](https://github.com/department-of-veterans-affairs/vets-website/pull/20435) with this removed.

Reply:
Integrating this feature into the Web Component via: https://github.com/department-of-veterans-affairs/component-library/pull/316